server/server/src/main/kotlin/org/jetbrains/bazel/server/sync/languages/LanguagePlugin.kt

```diff
@@ -20,6 +23,13 @@ abstract class LanguagePlugin<T : LanguageData> {
 
   open fun dependencySources(targetInfo: BspTargetInfo.TargetInfo, dependencyGraph: DependencyGraph): Set<Path> = emptySet()
 
+  open fun resolveBuilderPath(targetInfo: BspTargetInfo.TargetInfo): String? = null
+
+  open fun resolveBuilderArgs(targetInfo: BspTargetInfo.TargetInfo): List<String> = emptyList()
+
+
+  open fun prepareFastBuild(module: Module, params: FastBuildParams): FastBuildCommand? = null
+
```

Comment:
This is all JVM-specific anyway, so I don't think you have to put it into `LanguagePlugin`. I think those three methods could just be its own class.